cupy/linalg/einsum.py

```diff
@@ -291,6 +293,30 @@ def _flatten_transpose(a, axeses):
     )
 
 
+def _use_cutensor(dtype0, sub0, dtype1, sub1, batch_dims, contract_dims):
+    if not cupy.cuda.cutensor_enabled:
+        return False
+    if dtype0 != dtype1:
+        return False
+    if dtype0 not in (cupy.float32, cupy.float64,
+                      cupy.complex64, cupy.complex128):
+        return False
+    if len(sub0) == 0 or len(sub1) == 0:
+        return False
+    if (len(contract_dims) >= 1 and (sub0[-1] in batch_dims or
+                                     sub1[-1] in batch_dims)):
```

Comment:
Is the condition for performance?

Reply:
That is because the conditions above are not supported by cuTensor.

Reply:
I just noticed that no matmul nor cutensor is necessary when either/both of inputs is/are scalar...